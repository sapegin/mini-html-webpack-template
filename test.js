const compiler = require('@webpack-contrib/test-utils')
const MiniHtmlWebpackPlugin = require('mini-html-webpack-plugin')

const getConfig = (options, config = {}) =>
  Object.assign(
    {
      entry: ['./index.js'],
      plugins: [
        new MiniHtmlWebpackPlugin({
          context: options,
          template: require('./')
        })
      ]
    },
    config
  )

test('common options', () => {
  return compiler(
    {},
    getConfig({
      title: 'common options',
      favicon: '/favicon.ico',
      container: 'root',
      trimWhitespace: true
    })
  ).then(result => {
    expect(result.compilation.assets['index.html']._value).toMatchSnapshot()
  })
})

test('advanced options', () => {
  return compiler(
    {},
    getConfig({
      lang: 'en',
      title: 'advanced options',
      head: {
        meta: [
          {
            name: 'description',
            content: 'mini-html-webpack-template'
          },
          {
            property: 'og:description',
            content: 'mini-html-webpack-template'
          }
        ],
        links: [
          {
            rel: 'icon',
            type: 'image/x-icon',
            href: '/favicon.ico'
          }
        ],
        scripts: [
          {
            async: '',
            type: 'text/javascript',
            src: '/bundle.js'
          }
        ],
        raw: '<style id="head-raw-string"></style>',
        raw: [
          '<style id="head-raw-array-1"></style>',
          '<style id="head-raw-array-2"></style>'
        ]
      },
      body: {
        raw: '<div id="body-raw-string"></div>',
        raw: [
          '<script id="body-raw-array-1"></script>',
          '<script id="body-raw-array-2"></script>'
        ]
      }
    })
  ).then(result => {
    expect(result.compilation.assets['index.html']._value).toMatchSnapshot()
  })
})
