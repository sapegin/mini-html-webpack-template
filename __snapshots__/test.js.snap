// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`advanced options 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"UTF-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
    <meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\">
    <title>advanced options</title>
    <meta name=\\"description\\" content=\\"mini-html-webpack-template\\">
    <meta property=\\"og:description\\" content=\\"mini-html-webpack-template\\">
    <link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"/favicon.ico\\">
    <style id=\\"head-raw-array-1\\"></style>
    <style id=\\"head-raw-array-2\\"></style>
    <script async=\\"\\" type=\\"text/javascript\\" src=\\"/bundle.js\\"></script>
  </head>
  <body>
    <script id=\\"body-raw-array-1\\"></script>
    <script id=\\"body-raw-array-2\\"></script>
    <script src=\\"runtime~main.js\\"></script><script src=\\"main.js\\"></script>
  </body>
</html>"
`;

exports[`common options 1`] = `"<!DOCTYPE html><html><head><meta charset=\\"UTF-8\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\"><meta http-equiv=\\"X-UA-Compatible\\" content=\\"ie=edge\\"><title>common options</title><link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"/favicon.ico\\"></head><body><div id=\\"root\\"></div><script src=\\"runtime~main.js\\"></script><script src=\\"main.js\\"></script></body></html>"`;
